extensions/wikia/PhalanxII/tests/PhalanxModelTest.php

```diff
@@ -193,6 +193,71 @@ public function testPhalanxNewFromType( $type, $content, $className, $methodName
 		}
 	}
 
+	/**
+	 * @param bool $isPhalanxExempt whether session user is exempt from Phalanx blocks
+	 * @dataProvider isOkForPhalanxExemptUserDataProvider
+	 */
+	public function testIsOkForPhalanxExemptUser( bool $isPhalanxExempt ) {
+		/** @var PHPUnit_Framework_MockObject_MockObject|User $userMock */
+		$userMock = $this->getMock( User::class, [ 'isAllowed' ] );
+		$userMock->expects( $this->exactly( 3 ) )
+			->method( 'isAllowed' )
+			->with( $this->equalTo( 'phalanxexempt' ) )
+			->willReturn( $isPhalanxExempt );
+
+		$this->mockGlobalVariable( 'wgUser', $userMock );
+
+		$phalanxUserModel = new PhalanxUserModel( $userMock );
+		$res = $phalanxUserModel->isOk();
+		$this->assertEquals( $isPhalanxExempt, $res );
+
+		$phalanxContentModel = new PhalanxContentModel( new Title() );
+		$res = $phalanxContentModel->isOk();
+		$this->assertEquals( $isPhalanxExempt, $res );
+
+		$phalanxTextModel = new PhalanxTextModel( 'To jest test' );
+		$res = $phalanxTextModel->isOk();
+		$this->assertEquals( $isPhalanxExempt, $res );
+	}
+
+	/**
+	 * @param bool $isInternalRequest whether this is internal Wikia request
+	 * @param bool $isSelfCheck whether target user to look up is same as session user
+	 * @dataProvider isOkForInternalRequestDataProvider
+	 */
+	public function testIsOkForInternalRequest( bool $isInternalRequest, bool $isSelfCheck ) {
+		/** @var PHPUnit_Framework_MockObject_MockObject|User $userMock */
+		$userMock = $this->getMock( User::class, [ 'isAllowed', 'getName' ] );
+		$userMock->expects( $this->any() )
+			->method( 'isAllowed' )
+			->with( $this->equalTo( 'phalanxexempt' ) )
+			->willReturn( false );
+
+		if ( $isSelfCheck ) {
+			$userMock->expects( $this->once() )
+				->method( 'getName' )
+				->willReturn( $this->app->wg->User->getName() );
+		}
```

Comment:
In real-life examples `getName` returns either "a real" user name (when logged-in) or an IP address (when an anon). Can we change test here so that mocked values will match ones used in production cases?